### PR TITLE
Fix llm tests transformers

### PR DESCRIPTION
**Pull request by pablomlago**
## Risk Highlight

- [ ] This PR includes code from another work (please detail).
- [ ] This PR contains API-breaking changes.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [ ] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

## Checklist

- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [ ] Updated any relevant tests, if applicable.
- [ ] No conflicts with destination `dev` branch.
- [ ] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

File: tests/brevitas_examples/test_llm.py [hunk 1]
```diff
@@ -404,28 +467,39 @@ def test_small_models_acc_pt_ge_2_4(caplog, acc_args_and_acc_pt_ge_2_4):
...
         if hasattr(args, 'rotation') and args.rotation == 'fx' and platform.system() == 'Windows':
             pytest.skip("Skipping dynamo + windows")
     float_ppl, quant_ppl, model = validate_args_and_run_main(args)
-    assert_layer_types(model, exp_layer_types)
+    # Naming of modules in the GraphModule generated by FX changes across transformers versions, e.g.
+    # (4.45.0)"L__self___model_layers_2_self_attn_k_proj" ->
+    # (4.46.0) 'L__self___model_layers_slice_None__2__None___0_self_attn_q_proj'
+    # Therefore, this check is skipped when rotation="fx".
+    if args.rotation != "fx":
+        assert_layer_types(model, exp_layer_types)
```
Comment (Giuseppe5) [inline tests/brevitas_examples/test_llm.py:501]: We remove completely the type check

File: tests/brevitas_examples/test_llm.py [hunk 2]
```diff
@@ -299,7 +313,13 @@ def test_small_models_acc_pt_ge_2_4(caplog, acc_args_and_acc_pt_ge_2_4):
                 "model.layers.0.self_attn.q_proj.input_quant.fused_activation_quant_proxy.tensor_quant":
                     "<class 'brevitas.core.quant.int.RescalingIntQuant'>",
                 "model.layers.0.self_attn.q_proj.weight_quant.tensor_quant":
-                    "<class 'brevitas.core.quant.int.RescalingIntQuant'>",}},
+                    "<class 'brevitas.core.quant.int.RescalingIntQuant'>",},
+            "exp_layer_types_count": {
+                "<class 'torch.nn.modules.linear.Linear'>": 1,  # LM Head
+                "<class 'brevitas.nn.quant_linear.QuantLinear'>":
+                    14,  # Q/K/V/O projs + Up/Gate/Down projs
+                "<class 'brevitas.core.quant.int.RescalingIntQuant'>": 28,
```
Comment (Giuseppe5) [inline tests/brevitas_examples/test_llm.py:321]: We should only count QuantLayers

Comment (Giuseppe5) [reply tests/brevitas_examples/test_llm.py:321]: I'll let @nickfraser have the last word on these parts of the review

File: tests/brevitas_examples/test_llm.py [hunk 3]
```diff
@@ -404,28 +467,39 @@ def test_small_models_acc_pt_ge_2_4(caplog, acc_args_and_acc_pt_ge_2_4):
...
-    yield args, exp_layer_types
+    yield args, exp_layer_types, exp_layer_types_count
 
 
-@pytest.mark.llm
-@requires_pt_ge('2.2')
 def test_small_models_quant_layer(caplog, layer_args):
     caplog.set_level(logging.INFO)
-    args, exp_layer_types = layer_args
+    args, exp_layer_types, exp_layer_types_count = layer_args
```
Comment (nickfraser) [inline tests/brevitas_examples/test_llm.py:489]: My intuition is to split this into 2 different tests, 1 which checks the types of specific layer names, another which does the layer count tests. If we want the same models / configs tested then let's be verbose about that.

File: tests/brevitas_examples/test_llm.py (hunk 1 above)
Comment (nickfraser) [inline tests/brevitas_examples/test_llm.py:501]: If it was 2 tests then this would not be required.

File: tests/brevitas_examples/test_llm.py [hunk 4]
```diff
@@ -404,28 +467,39 @@ def test_small_models_acc_pt_ge_2_4(caplog, acc_args_and_acc_pt_ge_2_4):
...
+    exp_layer_types_count = layer_dict["exp_layer_types_count"]
     del layer_dict["exp_layer_types"]
+    del layer_dict["exp_layer_types_count"]
     args.update(**layer_dict)
-    yield args, exp_layer_types
+    yield args, exp_layer_types, exp_layer_types_count
 
 
-@pytest.mark.llm
-@requires_pt_ge('2.2')
```
Comment (nickfraser) [inline tests/brevitas_examples/test_llm.py:418]: Why is this removed?

File: tests/brevitas_examples/test_llm.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 from argparse import Namespace
+from collections import defaultdict
```
Comment (nickfraser) [inline tests/brevitas_examples/test_llm.py:5]: Not used?

**Review by nickfraser — CHANGES_REQUESTED** [review]

File: tests/brevitas_examples/test_llm.py (hunk 4 above)
Comment (pablomlago) [reply tests/brevitas_examples/test_llm.py:418]: I removed it accidentally.

**Pull request merged**